device/cuda/src/finding/finding_algorithm.cu

```diff
@@ -344,71 +345,74 @@ finding_algorithm<stepper_t, navigator_t>::operator()(
                      in.begin() + n_candidates_per_step[it], out.begin());
     }
 
-    /*****************************************************************
-     * Kernel6: Build tracks
-     *****************************************************************/
-
-    // Get the number of tips
+    /*
+     * Kernel 6: Count track lengths
+     */
     auto n_tips_total = m_copy.get_size(tips_buffer);
 
-    // Create track candidate buffer
-    track_candidate_container_types::buffer track_candidates_buffer{
-        {n_tips_total, m_mr.main},
-        {std::vector<std::size_t>(n_tips_total,
-                                  m_cfg.max_track_candidates_per_track),
-         m_mr.main, m_mr.host, vecmem::data::buffer_type::resizable}};
-
-    m_copy.setup(track_candidates_buffer.headers)->ignore();
-    m_copy.setup(track_candidates_buffer.items)->ignore();
-
-    // Create buffer for valid indices
-    vecmem::data::vector_buffer<unsigned int> valid_indices_buffer(n_tips_total,
+    vecmem::data::vector_buffer<unsigned int> valid_tip_idx_buffer(n_tips_total,
                                                                    m_mr.main);
+    vecmem::data::vector_buffer<unsigned int> valid_tip_length_buffer(
+        n_tips_total, m_mr.main);
+    vecmem::unique_alloc_ptr<unsigned int> n_valid_tips_device =
+        vecmem::make_unique_alloc<unsigned int>(m_mr.main);
 
-    unsigned int n_valid_tracks = 0;
+    TRACCC_CUDA_ERROR_CHECK(cudaMemsetAsync(n_valid_tips_device.get(), 0,
+                                            sizeof(unsigned int), stream));
 
-    // @Note: nBlocks can be zero in case there is no tip. This happens when
-    // chi2_max config is set tightly and no tips are found
-    if (n_tips_total > 0) {
-        vecmem::unique_alloc_ptr<unsigned int> n_valid_tracks_device =
-            vecmem::make_unique_alloc<unsigned int>(m_mr.main);
-        TRACCC_CUDA_ERROR_CHECK(cudaMemsetAsync(n_valid_tracks_device.get(), 0,
-                                                sizeof(unsigned int), stream));
+    std::vector<unsigned int> valid_tips_length_host;
```

Comment:
Asynchronous copying will not work into an `std::vector`. 😦 You'll need to allocate the host memory using `vecmem::cuda::host_memory_resource` for that.

Reply:
Asynchronous memory copies are well defined on non-pinned memory! See the last note on the documentation for `cudaMemcpyAsync`.

Reply:
😕 Is that happening with some new CUDA / driver version? Because in the past NSys was definitely not reporting any asynchronicity without using pinned host memory. I'll need to see a profile to believe this...

Reply:
🤔 Maybe not right now. But once we'll "give up" the CPU thread during synchronization points, it will be advantageous to set up the transfer, then go and start doing other things, and finally return to the thread once the copy (and everything before it) is done.

So we might as well do it with future proofing in mind already now.

Reply:
I would still prefer this to use `m_mr.host`.

Reply:
This isn't currently possible because the jagged vector constructor in vecmem only accepts default-allocated `std::vector` capacities. If I try passing a PMR vector with pinned host memory, vecmem will not accept it.